### PR TITLE
Adding a check to make sure Metadata is read only if flag value is true.

### LINKED ISSUE #74785: Kubelet failing to start on Azure stack
What happened:
Currently starting Kubectl failing on Azure Stack.
What you expected to happen:
Kubectl should start on Azure Stack environment.

How to reproduce it (as minimally and precisely as possible):
Try to deploy Kubernets on Azure Stack environment..

Anything else we need to know?:
Environment:
Kubernetes version (use kubectl version): master
Cloud provider or hardware configuration:
OS (e.g: cat /etc/os-release):
Kernel (e.g. uname -a):
Install tools:
Others:
/sig azure
/assign

### PR DESCRIPTION
**Pull request by rjaini**
What type of PR is this?
/kind bug
What this PR does / why we need it:
Fix kubelet start issue on Azure Stack. 

Which issue(s) this PR fixes:
Fixes #74785 
Special notes for your reviewer:
Does this PR introduce a user-facing change?:

```release-note
Fix kubelet start failure issue on Azure Stack due to InstanceMetadata setting
```

/kind bug
/assign @feiskyer
cc @hongbin

### CODE REVIEW

**Comment by andyzhangx** [comment]
/release-note `Fix kubelet start failure issue on Azure Stack due to InstanceMetadata setting`

**Comment by andyzhangx** [comment]
Hi @rjaini , would you sign the CLA first? we are close to code freeze time on k8s 1.14 soon.

File: pkg/cloudprovider/providers/azure/azure_zones.go [hunk 1]
```diff
@@ -49,31 +50,39 @@ func (az *Cloud) GetZoneID(zoneLabel string) string {
...
+			FailureDomain: zone,
+			Region:        az.Location,
+		}, nil
+	}
+	// if UseInstanceMetadata is false, get Zone name by calling ARM
+	hostname, err := os.Hostname()
+	if err != nil {
+		return cloudprovider.Zone{}, fmt.Errorf("failure getting hostname from kernel")
+	}
+	return az.vmSet.GetZoneByNodeName(hostname)
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_zones.go]: please cast the hostname to lower cases, as lower case is always used for kubernetes node name.

**Comment by feiskyer** [comment]
Please also fix the panic in unit tests.

**Comment by feiskyer** [comment]
Let's add this into v1.14 milestone since it's a critial fix for AzureStack.

/milestone v1.14

File: pkg/cloudprovider/providers/azure/azure_zones.go (hunk 1 above)
Comment (rjaini) [reply pkg/cloudprovider/providers/azure/azure_zones.go]: cast it to lower case.

**Comment by andyzhangx** [comment]
Run ./hack/update-gofmt.sh @rjaini 
```
diff -u ./pkg/cloudprovider/providers/azure/azure_test.go.orig ./pkg/cloudprovider/providers/azure/azure_test.go
--- ./pkg/cloudprovider/providers/azure/azure_test.go.orig	2019-03-05 04:48:13.311502387 +0000
+++ ./pkg/cloudprovider/providers/azure/azure_test.go	2019-03-05 04:48:13.312502475 +0000
@@ -1676,7 +1676,7 @@
 func TestGetZone(t *testing.T) {
 	cloud := &Cloud{
 		Config: Config{
-			Location:           "eastus",
+			Location:            "eastus",
 			UseInstanceMetadata: true,
 		},
 	}

Run ./hack/update-gofmt.sh
```

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
@feiskyer PTAL, also. I can cherry pick this PR to older version when it's merged.

**Comment by andyzhangx** [comment]
@rjaini thanks for your contribution, just let us know if there is sth broken on Azure stack in the future.

**Review by feiskyer — COMMENTED** [review]
lgtm, please squash the commits.

**Comment by justaugustus** [comment]
@rjaini -- Please add a release note to this PR. You can see how to do that here: https://git.k8s.io/community/contributors/guide/release-notes.md

@andyzhangx suggested the following: `Fix kubelet start failure issue on Azure Stack due to InstanceMetadata setting`.

Please also squash your commits.

**Comment by andyzhangx** [comment]
/release-note "Fix kubelet start failure issue on Azure Stack due to InstanceMetadata setting"

**Comment by rjaini** [comment]
Squash all commits.
@andyzhangx You can merge the commit.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
@rjaini  Added the release note for you. The template in PR body should be used.

**Comment by justaugustus** [comment]
(waking up the boots)
/joke

**Pull request merged**